charms/worker/k8s/src/charm.py

```diff
@@ -1048,6 +1048,7 @@ def _death_handler(self, event: ops.EventBase):
             self._revoke_cluster_tokens(event)
         self.update_status.run()
         self._last_gasp()
+        snap_management(self, remove=True)
```

Comment:
it's possible after last_gasp, the node is still clustered.  

maybe `_last_gasp(...)` should return a value if it succeeded, and only purge when the `_last_gasp(..)` succeeds

Reply:
I've updated the code. Wdyt?